pandas/tests/window/test_rolling.py

```diff
@@ -408,53 +473,97 @@ def test_rolling_window_as_string(using_array_manager):
     df = DataFrame({"DateCol": days, "metric": data})
 
     df.set_index("DateCol", inplace=True)
-    result = df.rolling(window="21D", min_periods=2, closed="left")["metric"].agg("max")
-
-    expData = (
-        [np.nan] * 2
-        + [88.0] * 16
-        + [97.0] * 9
-        + [98.0]
-        + [99.0] * 21
-        + [95.0] * 16
-        + [93.0] * 5
-        + [89.0] * 5
-        + [96.0] * 21
-        + [94.0] * 14
-        + [90.0] * 13
-        + [88.0] * 2
-        + [90.0] * 9
-        + [96.0] * 21
-        + [95.0] * 6
-        + [91.0]
-        + [87.0] * 6
-        + [92.0] * 21
-        + [83.0] * 2
-        + [86.0] * 10
-        + [87.0] * 5
-        + [98.0] * 21
-        + [97.0] * 14
-        + [93.0] * 7
-        + [87.0] * 4
-        + [86.0] * 4
-        + [95.0] * 21
-        + [85.0] * 14
-        + [83.0] * 2
-        + [76.0] * 5
-        + [81.0] * 2
-        + [98.0] * 21
-        + [95.0] * 14
-        + [91.0] * 7
-        + [86.0]
-        + [93.0] * 3
-        + [95.0] * 20
-    )
+    result = df.rolling(window="21D", min_periods=2, closed="left", center=center)[
+        "metric"
+    ].agg("max")
+
+    if center:
+        expected_data = (
+            [88.0] * 7
+            + [97.0] * 9
+            + [98.0]
+            + [99.0] * 21
+            + [95.0] * 16
+            + [93.0] * 5
+            + [89.0] * 5
+            + [96.0] * 21
+            + [94.0] * 14
+            + [90.0] * 13
+            + [88.0] * 2
+            + [90.0] * 9
+            + [96.0] * 21
+            + [95.0] * 6
+            + [91.0]
+            + [87.0] * 6
+            + [92.0] * 21
+            + [83.0] * 2
+            + [86.0] * 10
+            + [87.0] * 5
+            + [98.0] * 21
+            + [97.0] * 14
+            + [93.0] * 7
+            + [87.0] * 4
+            + [86.0] * 4
+            + [95.0] * 21
+            + [85.0] * 14
+            + [83.0] * 2
+            + [76.0] * 5
+            + [81.0] * 2
+            + [98.0] * 21
+            + [95.0] * 14
+            + [91.0] * 7
+            + [86.0]
+            + [93.0] * 3
+            + [95.0] * 29
+            + [77.0] * 2
+        )
+
+    else:
```

Comment:
if doing this pls paramterize outside the function

Reply:
Done!